rtos/TARGET_CORTEX/mbed_boot.c

```diff
@@ -507,6 +507,41 @@ void _mutex_free(mutex *m) {
 
 }
 
+#if (defined(__ARMCC_VERSION) && (__ARMCC_VERSION >= 6010050))
+
+// Check if Kernel has been started
+static uint32_t os_kernel_is_active (void) {
+  static uint8_t os_kernel_active = 0U;
+
+  if (os_kernel_active == 0U) {
+    if (osKernelGetState() > osKernelReady) {
+      os_kernel_active = 1U;
+      return 1U;
+    }
+    return 0U;
+  } else {
+    return 1U;
+  }
+}
+
+// Acquire mutex
+__USED void _mutex_acquire(mutex *m)
+{
+    if (os_kernel_is_active()) {
```

Comment:
Why do we need to check that?

Reply:
@bulislaw This code is copied from `mbed-os/rtos/TARGET_CORTEX/rtx5/RTX/Source/rtx_lib.c`. Maybe its original owner could comment it well. I guess it is to guard from e.g. C/C++ initialization code during which kernel is not ready yet.